PDEP0004: implementation (#49024)
pandas-dev/pandas: doc/source/user_guide/io.rst

Inferring datetime format
+++++++++++++++++++++++++

> Here are some examples of datetime strings that can be guessed (all

Author: good one - added!

Inferring datetime format
+++++++++++++++++++++++++

If you try to parse a column of date strings, pandas will attempt to guess the format
from the first non-NaN element, and will then parse the rest of the column with that
format.

.. ipython:: python

   df = pd.read_csv(
       "foo.csv",
       index_col=0,
       parse_dates=True,
   )
   df

In the case that you have mixed datetime formats within the same column, you'll need to

> first read it in the file, and then apply :func:`to_datetime` to each element.

Reviewer: ```suggestion
first read it in as an object dtype then apply :func:`to_datetime` to each element.
```

Would also be helpful to add an example
